integration/components/image/image_test.go

```diff
@@ -0,0 +1,196 @@
+
+	commonImage "github.com/inspektor-gadget/inspektor-gadget/cmd/common/image"
+	. "github.com/inspektor-gadget/inspektor-gadget/integration"
+)
+
+func TestImage(t *testing.T) {
+	// ensure that experimental is enabled
+	os.Setenv("IG_EXPERIMENTAL", "true")
+
+	// start registry
+	r := StartRegistry(t, "test-image-registry")
+	t.Cleanup(func() {
+		r.Stop(t)
+	})
+	if len(r.PortBindings()) == 0 || r.PortBindings()["5000/tcp"] == nil {
+		t.Fatal("registry port not exposed")
+	}
+	pb := r.PortBindings()["5000/tcp"][0]
+	registryAddr := fmt.Sprintf("%s:%s", pb.HostIP, pb.HostPort)
+
+	testReg := "reg.com"
+	testRepo := "repo1"
+	testTag := "tag1"
+	testLocalImage := fmt.Sprintf("%s/%s:%s", testReg, testRepo, testTag)
+	testRegistryImage := fmt.Sprintf("%s/%s:%s", registryAddr, testRepo, testTag)
+
+	// ensure all images are removed
+	t.Cleanup(func() {
+		// remove local image
+		cmd := commonImage.NewRemoveCmd()
+		cmd.SetArgs([]string{testLocalImage})
+		cmd.Execute()
+
+		// remove registry image
+		cmd = commonImage.NewRemoveCmd()
+		cmd.SetArgs([]string{testRegistryImage})
+		cmd.Execute()
+	})
+
+	type testCase struct {
+		name           string
+		cmd            *cobra.Command
+		args           []string
+		expectedStdout []string
+		expectedStderr []string
+	}
+
+	testCases := []testCase{
+		{
+			name: "build",
+			cmd:  commonImage.NewBuildCmd(),
+			args: []string{
+				"--builder-image", *testBuilderImage, "--tag", testLocalImage, "../../../gadgets/trace_open",
+			},
+			expectedStdout: []string{
+				fmt.Sprintf("Successfully built %s", testLocalImage),
+			},
+		},
+		{
+			name: "list",
+			cmd:  commonImage.NewListCmd(),
+			args: []string{},
+			expectedStdout: []string{
+				testRepo,
+				testTag,
+			},
+		},
+		{
+			name: "tag",
+			cmd:  commonImage.NewTagCmd(),
+			args: []string{testLocalImage, testRegistryImage},
+			expectedStdout: []string{
+				fmt.Sprintf("Successfully tagged with %s", testRegistryImage),
+			},
+		},
+		{
+			name: "push",
+			cmd:  commonImage.NewPushCmd(),
+			args: []string{testRegistryImage, "--insecure"},
+			expectedStdout: []string{
+				fmt.Sprintf("Successfully pushed %s", testRegistryImage),
+			},
+		},
+		{
+			name: "push-invalid-image",
+			cmd:  commonImage.NewPushCmd(),
+			args: []string{"unknown", "--insecure"},
+			expectedStderr: []string{
+				"failed to resolve ghcr.io/inspektor-gadget/gadget/unknown:latest: not found",
+			},
+		},
+		{
+			name: "push-unknown-tag",
+			cmd:  commonImage.NewPushCmd(),
+			args: []string{fmt.Sprintf("%s/%s:%s", registryAddr, testRepo, "unknown"), "--insecure"},
+			expectedStderr: []string{
+				fmt.Sprintf("%s/%s:%s: not found", registryAddr, testRepo, "unknown"),
+			},
+		},
+		{
+			name: "remove-local",
+			cmd:  commonImage.NewRemoveCmd(),
+			args: []string{testLocalImage},
+			expectedStdout: []string{
+				fmt.Sprintf("Successfully removed %s", testLocalImage),
+			},
+		},
+		{
+			name: "remove-registry",
```

Comment:
remove-tagged? (for a second I thought it was about removing the image from the registry)